src/hyperlink/_url.py

```diff
@@ -29,16 +29,15 @@
 except ImportError:  # Python 2
     from collections import Mapping
 
-# Note: IDNAError is a subclass of UnicodeError
-from idna import encode as idna_encode, decode as idna_decode, IDNAError
+from idna import encode as idna_encode, decode as idna_decode
 
 
 if inet_pton is None:
     # based on https://gist.github.com/nnemkin/4966028
     # this code only applies on Windows Python 2.7
     import ctypes
 
-    class _sockaddr(ctypes.Structure):
+    class _sockaddr(ctypes.Structure):  # noqa: N801
```

Comment:
`N801` is "you should use ClassNameCaps, yo", which… actually I guess we can do that here?

Reply:
The leading `"_"` is probably unnecessary here, since the whole module is private.

Reply:
Changed to `SockAddr`

Reply:
I think I just left it cased that way as an homage more than anything. It's fine titlecased.